### PR TITLE
Space Dragon Nerfs and Adjustments

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR nerfs and adjusts Space Dragon following some recent discussions on the matter and the apparent unbalanced state Space Dragon is in.  The changes are as follows:

- Max Health nerfed from 400 to 320
- Space Dragon now takes 50% of stamina damage, and will slow down the more stamina damage is inflicted.  Space Dragon slowly recovers from any stamina damage dealt
- Space Dragon now only heals 25% of the max health of a consumed victim as opposed to 50%
- Space Dragon's fire breath speed has been slowed down to pre-buff speed, which is a 33% nerf to the current speed (the fire itself moves slower again)
- Additional carp spawns now only come in 60-second intervals as opposed to 30-second ones.  This nerfs the total amount of carps Space Dragon can achieve before victory down to 18, where it was previously 33 (technically 17 and 32 if you consider the last carp of the third rift to be forfeit).
- Players cannot respawn from a carp rift they already spawned from unless there are 2 or more carp spawns on said rift.
- Fixed a bug where carp rifts without an owner would error when spawning in carp, letting them spawn in infinite carp.

I also wanted to include some additional barriers to prevent SD from using specific rooms as portal locations in this PR as well (specifically everything related to the AI Sat), however, I couldn't find a good way to go about it and wanted to get these changes out ASAP, so that will be on the backburner for now.

## Why It's Good For The Game

Space Dragon's currently overpowered status has been recently highlighted by Dynamic's apparent enjoyment in picking the big lug, so it's best we keep things fair and balanced and make sure the crew can still defeat the dragon regardless of other balance changes.

## Changelog
:cl:
balance: Space Dragon has received some nerfs to make fighting it and its carps fairer to the crew.
fix: Fixed a bug where carp rifts without an owner could spawn infinite carp.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
> Space Dragon now takes stamina damage, and will slow down the more stamina damage is inflicted. Space Dragon slowly recovers from any stamina damage dealt

bruh

> Additional carp spawns now only come in 60-second intervals as opposed to 30-second ones. This nerfs the total amount of carps Space Dragon can achieve before victory down to 18, where it was previously 33 (technically 17 and 32 if you consider the last carp of the third rift to be forfeit).

bruuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuh

**Comment by ATH1909** [comment]
the other changes look good, though. I think the main thing that made space dragons spawn was their ability to undo all of a competent player's hard work by just feasting on a single duffel bag wearer who ran in with an emergency oxygen tank, not their carp swarms. if anything, their carp spawning rate is too slow currently, not too fast.

**Comment by ATH1909** [comment]
fulp shitsec

**Comment by Mothblocks** [comment]
lethals are already the solution to every problem in this game, making them take damage from disablers etc is not moving the status quo worse at all

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -667,16 +670,25 @@
 /obj/structure/carp_rift/proc/summon_carp(mob/user)
 	if(carp_stored <= 0)//Not enough carp points
 		return FALSE
+	var/is_listed = FALSE
+	for(var/ckey in ckey_list)
+		if(user.ckey == ckey)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:675]: Just use `if (user.ckey in ckey_list)`?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by AndroidMK2** [comment]
This is quite the bruh moment. Lethals already exist for dealing with threats like this.

**Comment by Theta0006** [comment]
carp are easy to kill the only danger of them is if they cut the SMs out put pipe but the ability to stun the dragon is dumb why is something made for humans effective on a dragon

**Comment by Mothblocks** [comment]
i blame the carps that constantly respawned to blow up welding tanks

**Comment by moocowswag** [comment]
I think the problem with carps is they are only good at avoiding direct combat and just causing mass sabotage to the station, breaking apcs and external windows / airlocks in order to help the dragon out as much as possible without actually fighting anyone.

And making carps spawn less along with only letting one player spawn from the same rift under most conditions will mean that their live`s are more valuable and further enforce this as the optimal playstyle.

**Comment by ATH1909** [comment]
> carp are easy to kill the only danger of them is if they cut the SMs out put pipe but the ability to stun the dragon is dumb why is something made for humans effective on a dragon

tbf, the dragon won't get stunned from disabler fire, it'll just get slowed down.

**Comment by ATH1909** [comment]
> i blame the carps that constantly respawned to blow up welding tanks

and you think that they won't still do that with this PR?

**Comment by IndieanaJones** [comment]
> and you think that they won't still do that with this PR?

It'll encourage them to do it less knowing that their life is more valuable due to fewer carps overall and no chance of respawning until a new rift.

If you want t to argue the conclusions that were came to here, go argue it in the respective forum thread: https://tgstation13.org/phpBB/viewtopic.php?f=10&t=29452

**Comment by Mothblocks** [comment]
i think there won't be one after another after another after another yes

and if i i still see it then we will nerf it further

**Comment by adamkad1** [comment]
Sounds like ided tbh, all you need is COMPETENT PEOPLE to deal with dragon. Or just, you know, bomb its rift. People do that a lot and dragon can do jack shit to counter that

**Pull request merged**